blue_bringup/scripts/gripper_calibration_service.py

```diff
@@ -0,0 +1,149 @@
+        gripper_controller_response = switch_controller(['blue_controllers/gripper_torque_controller'], [], 2)
+
+        # Close the gripper
+        cmd = Float64MultiArray()
+        cmd.data = [5.0]
+        for i in range(5):
+            cmd_pub.publish(cmd)
+            rospy.sleep(0.1)
+
+        # Wait until gripper is closed/stops moving
+        gripper_state = get_gripper_state(motor_names)
+        current_position = gripper_state + 10
+        while not rospy.is_shutdown() and np.abs(current_position - gripper_state) > 1e-5:
+            cmd_pub.publish(cmd)
+            current_position = gripper_state
+            gripper_state = get_gripper_state(motor_names)
+
+        # Stop applying force
+        cmd.data = [0.0]
+        for i in range(5):
+            cmd_pub.publish(cmd)
+            rospy.sleep(0.1)
+
+        # Calibrate gripper to closed position
+        # TODO: don't hardcode this value -- read from yaml? URDF?
+        response = gripper_position_calibration(1.0)
+
+        if response.success:
+            rospy.loginfo("Gripper position calibration succeeded!")
+        else:
+            rospy.logerr("Gripper position calibration failed!")
+
+        # Switch out controllers to reset position
+        gripper_controller_response = switch_controller(['blue_controllers/gripper_controller'], ['blue_controllers/gripper_torque_controller'], 2)
+
+        client = actionlib.SimpleActionClient(
+            "blue_controllers/gripper_controller/gripper_cmd",
+            GripperCommandAction,
+        )
+
+        # Wait 10 Seconds for the gripper action server to start or exit
+        if not client.wait_for_server(rospy.Duration(10.0)):
+            rospy.logerr("Exiting - Gripper Action Server Not Found")
+            rospy.signal_shutdown("Action Server not found")
+            sys.exit(1)
+
+        goal = GripperCommandGoal()
+        goal.command.position = 0.0
+        goal.command.max_effort = 6.0
+        client.send_goal(goal)
+        client.wait_for_result()
+
+        # Clear gripper controllers
+        gripper_controller_response = switch_controller([], ['blue_controllers/gripper_controller'], 2)
+        rospy.loginfo("Gripper Calibration Succeeded!")
+
+        # Unload controllers that were loaded
+        for controller in unload_list:
+            unload_controller(controller)
+
+
+    except rospy.ServiceException as e:
+        rospy.logerr("Joint startup calibration failed: %s" % e)
+        return TriggerResponse(False, "Fail")
+
+    return TriggerResponse(True, "Succsess")
+
+if __name__ == "__main__":
+
+    rospy.init_node("simple_startup_calibration")
+
+    # Read startup angles from parameter server
+    rospy.loginfo("Reading desired joint angles...")
+    motor_names = rospy.get_param("blue_hardware/motor_names")
+
+    # Wait for calibration service to come up
+    rospy.loginfo("Waiting for calibration service...")
```

Comment:
comment + log text should be updated, since we're now also waiting for the controller manager